content/webapp/pages/guides/exhibitions/[id]/[type]/[stop]/index.tsx

```diff
@@ -0,0 +1,312 @@
+import { GetServerSideProps } from 'next';
+import Space from '@weco/common/views/components/styled/Space';
+import styled from 'styled-components';
+import {
+  ExhibitionGuideType,
+  isValidType,
+  GuideHighlightTour,
+} from '@weco/content/types/exhibition-guides';
+import { createClient } from '@weco/content/services/prismic/fetch';
+import { fetchExhibitionHighlightTour } from '@weco/content/services/prismic/fetch/exhibition-highlight-tours';
+import {
+  transformExhibitionHighlightTours,
+  transformGuideStopSlice,
+} from '@weco/content/services/prismic/transformers/exhibition-highlight-tours';
+import { serialiseProps } from '@weco/common/utils/json';
+import { getServerData } from '@weco/common/server-data';
+import { exhibitionGuideLd } from '@weco/content/services/prismic/transformers/json-ld';
+import { JsonLdObj } from '@weco/common/views/components/JsonLd/JsonLd';
+import { looksLikePrismicId } from '@weco/common/services/prismic';
+import { AppErrorProps } from '@weco/common/services/app';
+import { setCacheControl } from '@weco/content/utils/setCacheControl';
+import { isNotUndefined } from '@weco/common/utils/type-guards';
+import PageLayout from '@weco/common/views/components/PageLayout/PageLayout';
+import { pageDescriptions } from '@weco/common/data/microcopy';
+import { createPrismicLink } from '@weco/common/views/components/ApiToolbar';
+import { Container } from '@weco/common/views/components/styled/Container';
+import PrismicImage from '@weco/common/views/components/PrismicImage/PrismicImage';
+import { getCrop } from '@weco/common/model/image';
+import ImagePlaceholder, {
+  placeholderBackgroundColor,
+} from '@weco/content/components/ImagePlaceholder/ImagePlaceholder';
+import AudioPlayer from '@weco/content/components/AudioPlayer/AudioPlayer';
+import VideoEmbed from '@weco/common/views/components/VideoEmbed/VideoEmbed';
+import Icon from '@weco/common/views/components/Icon/Icon';
+import {
+  map,
+  audioDescribed,
+  britishSignLanguage,
+  cross,
+  arrow,
+} from '@weco/common/icons';
+
+type Props = {
+  jsonLd: JsonLdObj;
+  type: ExhibitionGuideType;
+  currentStop: GuideHighlightTour;
+  exhibitionGuideId: string;
+  exhibitionTitle: string;
+  stopNumber: number;
+  totalStops: number;
+};
+
+export const getServerSideProps: GetServerSideProps<
+  Props | AppErrorProps
+> = async context => {
+  setCacheControl(context.res);
+  const { id, type, stop } = context.query;
+
+  if (!looksLikePrismicId(id) || !isValidType(type)) {
+    return { notFound: true };
+  }
+
+  const client = createClient(context);
+
+  // TODO: get exhibitionHighlightTourQuery from localStorage if possible
+
+  const exhibitionHighlightTourQuery = await fetchExhibitionHighlightTour(
+    client,
+    id
+  );
+
+  if (isNotUndefined(exhibitionHighlightTourQuery)) {
+    const serverData = await getServerData(context);
+
+    if (isNotUndefined(exhibitionHighlightTourQuery)) {
```

Comment:
This might be my own ignorance, but why is there a second check for `isNotUndefined(exhibitionHighlightTourQuery)`? 

Would missing the second check flow the same?
```
if (isNotUndefined(exhibitionHighlightTourQuery)) {
    const serverData = await getServerData(context);

    const exhibitionHighlightTour = transformExhibitionHighlightTours(
        exhibitionHighlightTourQuery
    );
```

Reply:
This is because I've copied and pasted code that had more checks that I have since removed, but failed to remove the outer check. Nice catch, thanks!